app/src/components/app.rs

```diff
@@ -247,15 +249,13 @@ impl App {
                     id="dropdownUser"
                     data-bs-toggle="dropdown"
                     aria-expanded="false">
-                    <svg xmlns="http://www.w3.org/2000/svg"
-                      width="32"
-                      height="32"
-                      fill="currentColor"
-                      class="bi bi-person-circle"
-                      viewBox="0 0 16 16">
-                      <path d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"/>
-                      <path fill-rule="evenodd" d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm8-7a7 7 0 0 0-5.468 11.37C3.242 11.226 4.805 10 8 10s4.757 1.225 5.468 2.37A7 7 0 0 0 8 1z"/>
-                    </svg>
+                    {
+                      if let Some((user_id, _)) = &self.user_info {
+                        html! {
+                          <Avatar username=user_id.clone() width=32 height=32 />
```

Comment:
~You want to be able to send a message to the `Avatar` component to reload when the avatar changes. As it turns out, this is rather tricky. One solution would be with: https://dev.to/davidcks/how-to-send-a-message-to-a-child-component-in-yew-503~

~Then once you have the component link of the avatar child, you can send a message to the child. Not ideal, but that's the best we can have, I think.~

Actually, now that I think about it, we can just change the properties of Avatar as a message, since it just needs to reload. By setting a property like "last avatar changed time", the Avatar child can re-render (and re-fetch) when it receives this. Much simpler.

The other part of the story is to pass a callback that will send a message to the App with the last avatar changed time. You pass that callback down into UserDetails/UserDetailsForm components, and in the form, if the avatar changes, you call the callback.

Reply:
Actually, the proper solution for this is to use agents: https://yew.rs/docs/0.18.0/concepts/agents

Reply:
Thanks for this! I'll try to make the time soon for using that

Reply:
I'm working on an update of the frontend to yew 0.19 or 0.20, so you might as well wait until that's done, since that'll change a bunch of things.

Reply:
Should this still become an agent, or is this something that would work well with yew 0.19 context?

Reply:
I think it should be an agent. But I've never used them :)

Reply:
Actually, we can also have a look at the contexts: https://yew.rs/docs/next/concepts/contexts

Reply:
Used agents after all, a little smoother when i went down both rabbit holes